stdlib/public/core/StringGutsRangeReplaceable.swift

```diff
@@ -512,34 +508,19 @@ extension _StringGuts {
         oldBounds.lowerBound,
         oldBounds.upperBound &+ newRange.count &- oldRange.count))
 
-    // Update `startIndex` if necessary. The replacement may have invalidated
```

Comment:
I don't remember all this code (e.g. character alignment). @lorentey or @Azoy can you review this PR as well?